### ISSUE TITLE
[Bug] [IN-17625] AREnvironmentProbeManager.OnDestroy triggers application crash

### ISSUE THREAD

**Issue by Phantomxm2021** [Unity-Technologies/arfoundation-samples#1003]
**Describe the bug**

Instantiate a game object with an AREnvironment prob component and destroys it. Then loading other scenes (without AR)... 

Bump, **crash**.

```log
NullReferenceException: Object reference not set to an instance of an object.
  at UnityEngine.XR.ARFoundation.AREnvironmentProbeManager.OnDestroy () [0x00000] in <00000000000000000000000000000000>:0 
```

**Smartphone (please complete the following information):**
 - Device: iPhone13 pro max
 - OS: 15.6.1
 - Unity version: Unity 2021.3.8f1
 - ARFoundation version: 5.0.0-pre.13

**Comment by DavidMohrhardt** [comment]
Please file an official bug and link the number so we can get it. https://unity3d.com/unity/qa/bug-reporting

**Comment by Phantomxm2021** [comment]
> Please file an official bug and link the number so we can get it. https://unity3d.com/unity/qa/bug-reporting

bug case id: IN-17625

**Comment by andyb-unity** [comment]
Fixed in 5.0.7: https://docs.unity3d.com/Packages/com.unity.xr.arfoundation@5.0/changelog/CHANGELOG.html
